doc/src/tutorial/codegen.rst

```diff
@@ -1,95 +1,105 @@
+powers in C. ``x**2`` prints as ``pow(x, 2)`` instead of ``x*x``.  Other
+optimizations (like mathematical simplifications) should happen before the code
+printers.
 
-The code printers don't print optimal code in many cases. 
-An example of this is powers in C. x**2 prints as pow(x, 2) instead of x*x. 
-Other optimizations (like mathematical simplifications) should happen 
-before the code printers.
+Currently, :py:func:`sympy.cse` is not applied automatically anywhere in this
+chain. It ideally happens at the codegen level, or somewhere above it.
 
-cse is not applied anywhere in this chain. It ideally happens at the 
-codegen level, or somewhere above it.
+We will iterate through the levels below.
 
-We will iterate through the levels, bottom up.
+The following three lines will be used to setup each example::
+
+    >>> from sympy import *
+    >>> init_printing(use_unicode=True)
+    >>> from sympy.abc import a, e, k, n, r, t, x, y, z, T, Z
+    >>> from sympy.abc import beta, omega, tau
+    >>> f, g = symbols('f, g', cls=Function)
 
 Code printers (sympy.printing)
 ------------------------------
-This is where the meat of code generation is, the translation of SymPy
-expressions to specific code.Supported languages are C (``ccode``), Fortran 95 (``fcode``), 
-Javascript (``jscode``), Mathematica (``mathematica_code``), Octave/Matlab (``octave_code``), 
-Python (print_python, which is actually more like a lightweight version 
-of codegen for Python, and ``lambdarepr``, which supports many libraries 
-(like NumPy), and theano (``theano_code``).The code printers 
-are special cases of the other prints in SymPy (str printer, pretty printer, etc.).
-
-An important distinction is that the code printer has to deal with 
-assignments (using the ``sympy.printing.codeprinter.Assignment`` object).This serves
-as building blocks for the code printers and hence the ``codegen`` module.
-An example that shows the use of ``Assignment``::
-
-    >>> from sympy import symbols, MatrixSymbol, Matrix
+
+This is where the meat of code generation is; the translation of SymPy
+expressions to specific languages. Supported languages are C
+(:py:func:`sympy.printing.ccode.ccode`), Fortran 95
+(:py:func:`sympy.printing.fcode.fcode`), Javascript
+(:py:func:`sympy.printing.jscode.jscode`) , Mathematica
+(:py:func:`sympy.printing.mathematica.mathematica_code`), Octave/Matlab
+(:py:func:`sympy.printing.octave.octave_code`), Python (print_python, which is
+actually more like a lightweight version of codegen for Python, and
+:py:func:`sympy.printing.lambdarepr.lambdarepr`, which supports many libraries
+(like NumPy), and theano
+(:py:func:`sympy.printing.theanocode.theano_function`). The code printers are
+special cases of the other prints in SymPy (str printer, pretty printer, etc.).
+
+An important distinction is that the code printer has to deal with assignments
+(using the ``sympy.printing.codeprinter.Assignment`` object).This serves as
```

Comment:
I think to make it a link you need to use :ref:\`sympy.printing.codeprinter.Assignment\` (my rst is rusty so that may not be exactly the right thing)